src/keystore/persistence/EncryptedPersistence.ts

```diff
@@ -0,0 +1,82 @@
+import { Persistence } from './interface'
+import eccrypto, { Ecies } from 'eccrypto'
+
+const assertEciesLengths = (ecies: Ecies): void => {
+  if (ecies.iv.length !== 16) {
+    throw new Error('Invalid iv length')
+  }
+  if (ecies.ephemPublicKey.length !== 65) {
+    throw new Error('Invalid ephemPublicKey length')
+  }
+  if (ecies.ciphertext.length < 1) {
+    throw new Error('Invalid ciphertext length')
+  }
+  if (ecies.mac.length !== 32) {
+    throw new Error('Invalid mac length')
+  }
+}
+
+const serializeEcies = (ecies: Ecies): Uint8Array => {
+  assertEciesLengths(ecies)
+  const { iv, ephemPublicKey, ciphertext, mac } = ecies
+  const result = new Uint8Array(
+    iv.length + ephemPublicKey.length + ciphertext.length + mac.length
+  )
+  result.set(iv, 0)
+  result.set(ephemPublicKey, iv.length)
+  result.set(ciphertext, iv.length + ephemPublicKey.length)
+  result.set(mac, iv.length + ephemPublicKey.length + ciphertext.length)
+  return result
+}
+
+const deserializeEcies = (data: Uint8Array): Ecies => {
+  const iv = data.slice(0, 16)
```

Comment:
From a crypto auditability/ readability POV the slice indexing here could be improved. Extracting the slice indexes into variables which are used for both Serialize and Deserialize would add symmetry and make mistakes easier to detect in future PRs

Reply:
Sounds reasonable

Reply:
Done